hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1214,24 +1215,13 @@ def shrink_pass(fn):
     """Decorates a function that is intended to be used as a shrink pass to
     give some uniform behaviour and logging."""
     @proxies(fn)
-    def run(self):
-        self.debug('Shrink Pass %s' % (fn.__name__,))
-        initial_shrinks = self.shrinks
-        initial_calls = self.calls
-        try:
-            fn(self)
-        finally:
-            calls = self.calls - initial_calls
-            shrinks = self.shrinks - initial_shrinks
-            self.debug((
-                'Shrink Pass %s completed. Made %d call%s and '
-                '%d shrink%s') % (
-                fn.__name__,
-                calls, 's' if calls != 1 else '',
-                shrinks, 's' if shrinks != 1 else '',
-            ))
-            if not self.discarding_failed:
-                self.remove_discarded()
+    def run(self, *args, **kwargs):
+        name = '%s(%s)' % (
+            fn.__name__, ', '.join(tuple(map(repr, args)) + tuple(sorted(
+                '%s=%r' % (k, v) for k, v in kwargs.items()))))
```

Comment:
I’ll note that this sort of pretty-printing is used in a couple of places. You don’t need to pull it out here, but commonising it might be nice.